pkg/prometheus/server/operator.go

```diff
@@ -88,15 +88,42 @@ type Operator struct {
 	reconciliations *operator.ReconciliationTracker
 	statusReporter  prompkg.StatusReporter
 
+	eventRecorder record.EventRecorder
+	options       controllerOptions
+}
+
+type controllerOptions struct {
```

Comment:
Do we need an intermediate struct? `o.options.scrapeConfigSupported` is a bit heavy to read.

Reply:
No, no need for it :)

I was just grouping things together, but no strong opinions here